cmd/multiclusterapp.go

```diff
@@ -679,45 +763,203 @@ func fromMultiClusterAppAnswers(answers []managementClient.Answer) map[string]st
 	return answerMap
 }
 
-func showMultiClusterApp(ctx *cli.Context) error {
-	if ctx.NArg() == 0 {
+func addMcappTargetProject(ctx *cli.Context) error {
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	app, input, err := proceedUpdateMultiClusterAppTargetsArgs(ctx, c)
+	if err != nil {
+		return err
+	}
+	return c.ManagementClient.MultiClusterApp.ActionAddProjects(app, input)
+}
+
+func deleteMcappTargetProject(ctx *cli.Context) error {
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	app, input, err := proceedUpdateMultiClusterAppTargetsArgs(ctx, c)
+	if err != nil {
+		return err
+	}
+	return c.ManagementClient.MultiClusterApp.ActionRemoveProjects(app, input)
+}
+
+func proceedUpdateMultiClusterAppTargetsArgs(ctx *cli.Context, c *cliclient.MasterClient) (*managementClient.MultiClusterApp, *managementClient.UpdateMultiClusterAppTargetsInput, error) {
+	if len(ctx.Args()) < 2 {
+		return nil, nil, cli.ShowSubcommandHelp(ctx)
```

Comment:
`cli.ShowSubcommandHelp(ctx)` is not an error, it prints the help for the command, so the error checks in `addMcappTargetProject` and `deleteMcappTargetProject` don't catch it which causes a panic further down. Check the length of args in the first function.